### ISSUE TITLE
crypto: provide simple way to add new curves 

### ISSUE THREAD

**Issue by tac0turtle** [cometbft/cometbft#2424]
### Summary

currently there is not a way to add new curves to comet unless you fork the software. This causes many forks for simple changes.

### Problem Definition

adding new curves forces users to fork the software. (unless im missing something)

### Proposal

add simple way to add new curves to be used for consensus key signing

Zondax is proposing an api to help with this in the sdk. we could look at hosting the new api in its own module to enable usage in comet as well

**Comment by tac0turtle** [comment]
to add on it would be nice to have the file system privval also take arbitrary curves. This way users dont also have to fork privval file once pluggable curves land

**Comment by cason** [comment]
I think we would need a method such as `RegisterKey` at some place.

Now one can implement the `crypto.PrivKey` and `crypto.PrivKey` interfaces, but not sure how to set the new key for use by Comet.

**Comment by tac0turtle** [comment]
i think the pubkey proto type may need some changing as the current design wouldnt allow register key without redoing the proto types

**Comment by melekes** [comment]
refs https://github.com/cometbft/cometbft/pull/2399
